### ISSUE TITLE
Failed to connect to server

### ISSUE THREAD

**Issue by volume4k** [splewis/get5-web#49]
After I installed get5-web I tried to connect to the CS:GO server which was running and had the get5 server plugins AND the get5_apistats plugins installed and running. At least it returned:
 `get5_web_avaliable
{
    "plugin_version": "0.5.1",
    "gamestate": 0,
    "avaliable": 1
}` 

But it just says: `Failed to connect to server` What could be the issue? 
Note: The ports are both matching either. I tried both... the localhost (127.0.0.1) and the public IP. The rcon password is right also.

**Comment by Technoblazed** [comment]
Check that all ports are available, where did it say `Failed to connect to server` as well.

**Comment by volume4k** [comment]
It said this only on the web interface, but I can connect and play CS:GO on the server as usually.

**Comment by Technoblazed** [comment]
Could you test the rcon connection from another location, it may be that the port is being blocked elsewhere.

**Comment by volume4k** [comment]
How can I do that? Because I'm not quite sure how to test it... just via CS:GO itself (VPN)? Or how do I do it? 
Here is a screen of the error: http://prntscr.com/exwjsw

**Comment by Technoblazed** [comment]
And you tested it with the public IP as well correct?

**Comment by volume4k** [comment]
Yes of course I have! I just don't want to reveal the public one.

**Comment by volume4k** [comment]
I don't know why but it seems like the port 27015 is closed and I tried to open it but it wont change anything. Do you have an Idea what to do? I tried it with `iptables` so far.
Thankyou btw for your fast responses!

**Comment by Technoblazed** [comment]
Could you print the output of `iptables -L`, removing anything that clearly shouldn't be there.

**Comment by volume4k** [comment]
output of iptables -L :

Chain INPUT (policy ACCEPT)
target     prot opt source               destination

Chain FORWARD (policy ACCEPT)
target     prot opt source               destination

Chain OUTPUT (policy ACCEPT)
target     prot opt source               destination

**Comment by Technoblazed** [comment]
Well it's not that. Could you confirm that the gameserver is running on port 27015 for me? `netstat -tulpn | grep :27015` - Does this return a response?

**Comment by volume4k** [comment]
It returns the following:
`udp        0      0 0.0.0.0:27015           0.0.0.0:*                           2182/./srcds_linux`

**Comment by Technoblazed** [comment]
Your server isn't listening on the TCP port for CSGO, that command should return both a TCP and UDP version if the server is running correctly, could you paste your server start line here please :D

**Comment by volume4k** [comment]
screen -A -m -d -S csgo ./srcds_run -tickrate 128 +map de_dust2 +game_mode 0 +game_type 0 +sv_setsteamaccount E0F6B3A4103E29B01564520CA*******

**Comment by Technoblazed** [comment]
Could you add `-strictportbind -ip 0.0.0.0 -port 27015` to the end of that for me please :D

**Comment by volume4k** [comment]
Yeah I've done it but it still just returns:
 `udp 0 0 0.0.0.0:27015 0.0.0.0:* 2372/./srcds_linux`

**Comment by Technoblazed** [comment]
If you check the console by attaching to the screen, do you see any warnings in the console?

**Comment by volume4k** [comment]
I can't find any error regarding the ports :c 
https://pastebin.com/LBxQ0dh1

**Comment by volume4k** [comment]
`udp        0      0 0.0.0.0:26901           0.0.0.0:*                           2465/./srcds_linux
udp        0      0 0.0.0.0:27005           0.0.0.0:*                           2465/./srcds_linux
udp        0      0 0.0.0.0:27015           0.0.0.0:*                           2465/./srcds_linux
udp        0      0 0.0.0.0:27020           0.0.0.0:*                           2465/./srcds_linux`

That's what I have now.

**Comment by Technoblazed** [comment]
Well, it's 100% a TCP issue with your server. It's not IPtables because you have everything set to allow. Would you be able to login to a root/sudo user and run the original netstat command again, `netstat -tulpn | grep :27015`, curious if the process is already running under another user :thinking:

**Comment by volume4k** [comment]
This actually is a root allowed user and I don't have the "root" user unfortunately. But if I am doing `htop` after `sudo su` I still can see that only my user is running the CS:GO server or a server on :27015.

**Comment by Technoblazed** [comment]
I honestly have no idea why you're unable to bind to the port then, this isn't necessarily a get5 issue, but moreso a server configuration issue :thinking:

**Comment by volume4k** [comment]
Yea I think so to because from the get5 side all worked flawless. Maybe I'll find something within the config of the server because this is a proxmox virtual server which is hosted by OVH so the network config is different to AWS, DigitalOcean, MyLoc,... (a lot harder actually)
Thankyou for your help and for finding out that it might be a serverside issue 💯 👍

**Comment by Technoblazed** [comment]
I usually just use the linux dedis from OVH :man_shrugging:

**Comment by volume4k** [comment]
Yeah it's a proxmox (which is linux) dedi 🥇

**Comment by volume4k** [comment]
I think I need to go for a clean OS install :c 
What Linux would you recommend for hosting this server?

**Comment by Technoblazed** [comment]
Personally, any debian/redhat distros.

**Comment by volume4k** [comment]
Now I am on a new Ubuntu after failing for about 3 days trying to get the network configuration running but no luck so far. So we setup a new Ubuntu 17.04 and it finally worked. Then after a lot of installing issues I finally got to 'Add Server'... I typed all in correctly and it said unable to connect. After that I felt like: "No that isn't gonna happen again!?!" Then I tried to figure out what was wrong and after I've seen with your great `sudo netstat -tulpn | grep :27015` command that there is a tcp port and it was just binded to 127.0.0.1. So actually in the end `-strictportbind -ip 0.0.0.0 -port 27015` works ^^ 
Thankyou for your great support man <3
